core/src/main/java/com/graphhopper/routing/util/parsers/OSMMaxSpeedParser.java

```diff
@@ -56,9 +53,13 @@ public void createEncodedValues(EncodedValueLookup lookup, List<EncodedValue> li
     public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay way, boolean ferry, IntsRef relationFlags) {
         double maxSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed"));
 
+        RoadClass roadClass = RoadClass.find(way.getTag("highway", ""));
+        CountryRule countryRule = way.getTag("country_rule", null);
+        if (countryRule != null)
+            maxSpeed = countryRule.getMaxSpeed(roadClass, TransportationMode.CAR, maxSpeed);
```

Comment:
As noted in https://github.com/graphhopper/graphhopper/issues/2246#issuecomment-878078733 we should probably pass this down to the value extractor.

Reply:
Ok, see discussion in #2246.